manager/backend/api.py

```diff
@@ -1,65 +1,175 @@
+
+		return Response(authorData)
+
+class UserAPI(generics.RetrieveAPIView):
+
+	permission_classes = [
+		permissions.IsAuthenticated,
+	]
+
+	serializer_class = UserSerializer
+
+	def get_object(self):
+		return self.request.user
+
+
+
+
 class AuthorViewSet(viewsets.ModelViewSet):
-    
-    queryset = Author.objects.all()
+	"""
+	This class specifies the view for the Post objects. This will run methods to retrieve and edit DB rows and return correctly formatted HTTP responses
+	"""
+	# Specifies the query set on which this view can act
+	queryset = Author.objects.all()
+
+	# Specifies the permissions needed to access and modify the data
+	permission_classes = [
+		permissions.IsAuthenticated
+	]
+
+	# Specifies the serializer used to return the correctly formatted JSON response body
+	serializer_class = AuthorSerializer
+
+	# Specifies the lookup field to use the in the database
+	lookup_field = 'id'
+
+	def update(self, request, *args, **kwargs):
+		"""
+		This method will be called when a POST request is received for a specific author.
+		"""
+
+		# Code to extract specifically the author's id value from the url in the JSON request's id field
+		author_id = request.data["id"]
+		newID = author_id.split("/")[-2]
+
+		# Update the request object's data so the id field is correctly formatted to be found in the DB
+		request.data.update({"id": newID})
+
+		return super().update(request, *args, **kwargs)
+
+
+# Get Author API
+class LoginAPI(viewsets.ModelViewSet):
+
+	queryset = Author.objects.all()
+
+	permission_classes = [
+		#permissions.AllowAny
+		permissions.AllowAny
+	]
+
+	serializer_class = AuthorSerializer
 
-    permission_classes = [
-        permissions.AllowAny
-    ]
-    serializer_class = AuthorSerializer
 
-    lookup_field = 'id'
 
-# Register API
-class RegisterAPI(generics.GenericAPIView):
-  serializer_class = RegisterSerializer
-
-  def post(self, request, *args, **kwargs):
-    serializer = self.get_serializer(data=request.data)
-    serializer.is_valid(raise_exception=True)
-    user = serializer.save()
-    token = Token.objects.create(user=user)
-    userData = UserSerializer(user, context=self.get_serializer_context()).data
-    return Response({
-      "user": userData
-    })
 class PostViewSet(viewsets.ModelViewSet):
+	"""
+	This class specifies the view for the Post objects. This will run methods to retrieve and edit DB rows and return correctly formatted HTTP responses
+	"""
+
+	# Specifies the permissions required to access the data
+	permission_classes = [
+		permissions.AllowAny
+	]
+
+	# Specifies the field used for querying the DB
+	lookup_field = 'id'
+
+	# Specifies the serializer used to return a properly formatted JSON response body
+	serializer_class = PostSerializer
+
+	# Specifies the query set of Post objects that can be returned
+	queryset = Post.objects.all()
+
+	def list(self, request, author_id=None, id=None, *args, **kwargs):
```

Comment:
which one is for a POST request for creating a new post?

Reply:
I actually added this bit in an earlier commit I'm not sure why it shows as changed in this latest one, but anyways the ModelViewSet comes with action methods already implemented. I have them attached to HTTP requests in the urls.py. The only reason I overwrote the functionality on some of them is that the queries needed specific information. Currently, POST should call the create() method which is already implemented by Django and didn't need to be overwritten to function correctly.

Reply:
sounds good